### PR TITLE
Close #2054 Note that any Drush version lower than`12.4.3` will cause a conflict,  and add `az-entity-list:list` drush command.

### LINKED ISSUE #2054: Replicate uaqs_list_assets drush command for Quickstart 2
## Motivation
In an effort to make reporting site metrics easier we would like a way to easily list entity asset counts on sites using Quickstart 2.

## Proposed Resolution
Add a list assets drush command to list QS2 assets.

## Additional context
See https://bitbucket.org/ua_drupal/ua_quickstart/src/3bf9f9b4116144655178dde6c21c572273686411/modules/custom/uaqs_core/uaqs_core.drush.inc#lines-38:73

### PR DESCRIPTION
**Pull request by trackleft**
## Release Notes

```
- The new minimum version of Drush is now `12.4.3`
```

## Description
This Pull Request adds a drush command that lists how many entities of each bundle 

This Pull Request also does a little house keeping 

## Related issues
#2054 

## How to test

1. **In Probo open Advanced Functionality**

image

## 

2. **Then Click SSH Terminal to open the terminal**

In the terminal, you should be able to run:

```
drush az-entity-list:list
```

3. **Also test existing command, just to be sure that it runs, since it is moved in this PR.**. (This should be backwards-compatible if used as a service).

```
drush az-core-config-add-permissions
```

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [x] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_core/src/Commands/README.md [hunk 1]
```diff
@@ -1,29 +1,94 @@
-# AZ Core Config Commands
+# Quickstart Core Drush Commands
+AZ Core provides Drush commands to help with managing an AZ Quickstart website.
 
-AZ Core provides Drush commands to help with managing configuration on AZ Quickstart websites.
+### Requirements
 
-## Requirements
 - Drupal 9 or 10
```
Comment (joeparsons) [inline modules/custom/az_core/src/Commands/README.md:6]: Technically D9 is EOL now, so might as well just say 10+
```suggestion
- Drupal 10
```

Comment (joeparsons) [reply modules/custom/az_core/src/Commands/README.md:6]: Or, alternatively:
```suggestion
- Arizona Quickstart 2.10.0 or greater.
```

File: modules/custom/az_core/src/Commands/README.md [hunk 2]
```diff
@@ -1,29 +1,94 @@
...
 Constructs a new AZCoreConfigCommands object.
 
 ### addMissingPermissions()
 A custom Drush command to add missing installation profile permissions.
+
+## AZ Entity List
+
+Provides a drush command for listing entities.
+
+### Running the drush command
```
Comment (trackleft) [inline modules/custom/az_core/src/Commands/README.md:40]:
```suggestion
### Usage
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  paragraph             az_accordion                     0
+  paragraph             az_cards                         4
+  paragraph             az_contact                       0
+  paragraph             az_photo_gallery                 0
+  paragraph             az_splitscreen                   21
+  paragraph             az_text                          17
+  paragraph             az_text_background               9
+  paragraph             az_text_media                    14
+  paragraph             az_view_reference                4
+ --------------------- -------------------------------- -------
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:93]:
```suggestion
 ------------------------ -------------------------------- ------------------- ------- 
  Entity type              Bundle                           Provider            Count  
 ------------------------ -------------------------------- ------------------- ------- 
  block                    block                            block               47     
  block_content            az_custom_menu_block             block_content       3      
  block_content_type       block_content_type               block_content       4      
  config_snapshot          config_snapshot                  config_snapshot     76     
  crop_type                crop_type                        crop                8      
  editor                   editor                           editor              2      
  embed_button             embed_button                     embed               3      
  field_config             field_config                     field               85     
  field_storage_config     field_storage_config             field               63     
  file                     file                             file                41     
  filter_format            filter_format                    filter              5      
  google_tag_container     google_tag_container             google_tag          1      
  image_style              image_style                      image               27     
  linkit_profile           linkit_profile                   linkit              2      
  media_type               media_type                       media               5      
  media                    az_image                         media               35     
  media                    az_remote_video                  media               6      
  menu_link_content        menu_link_content                menu_link_content   34     
  metatag_defaults         metatag_defaults                 metatag             7      
  migration_group          migration_group                  migrate_plus        1      
  node_type                node_type                        node                4      
  node                     az_event                         node                23     
  node                     az_flexible_page                 node                22     
  node                     az_news                          node                5      
  node                     az_person                        node                13     
  path_alias               path_alias                       path_alias          144    
  redirect                 redirect                         redirect            1      
  responsive_image_style   responsive_image_style           responsive_image    1      
  search_page              search_page                      search              4      
  shortcut                 default                          shortcut            2      
  shortcut_set             shortcut_set                     shortcut            1      
  smart_date_format        smart_date_format                smart_date          4      
  smart_date_rule          smart_date_rule                  smart_date_recur    9      
  action                   action                           system              30     
  menu                     menu                             system              11     
  taxonomy_vocabulary      taxonomy_vocabulary              taxonomy            7      
  taxonomy_term            az_event_categories              taxonomy            23     
  taxonomy_term            az_news_tags                     taxonomy            4      
  taxonomy_term            az_page_categories               taxonomy            12     
  taxonomy_term            az_person_categories             taxonomy            19     
  taxonomy_term            az_person_categories_secondary   taxonomy            11     
  tour                     tour                             tour                3      
  user_role                user_role                        user                7      
  user                     user                             user                2      
  workflow                 workflow                         workflows           1      
  pathauto_pattern         pathauto_pattern                 pathauto            6      
  xmlsitemap               xmlsitemap                       xmlsitemap          1      
  view                     view                             views               22     
  paragraphs_type          paragraphs_type                  paragraphs          9      
  paragraph                az_cards                         paragraphs          4      
  paragraph                az_splitscreen                   paragraphs          21     
  paragraph                az_text                          paragraphs          17     
  paragraph                az_text_background               paragraphs          9      
  paragraph                az_text_media                    paragraphs          14     
  paragraph                az_view_reference                paragraphs          4      
  base_field_override      base_field_override              core                5      
  date_format              date_format                      core                13     
  entity_form_display      entity_form_display              core                29     
  entity_form_mode         entity_form_mode                 core                2      
  entity_view_display      entity_view_display              core                73     
  entity_view_mode         entity_view_mode                 core                47     
 ------------------------ -------------------------------- ------------------- ------- 

```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+Aliases: `ael`
+
+```
+lando drush az-entity-list:list
+```
+
+### Output
+
+The default output will looks like this.
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:63]:
```suggestion
The default output will count each entity type by bundle, and output in a table by default.

To output in other formats, use `drush az-entity-list:list --help`
```

File: modules/custom/az_core/src/Drush/Commands/AZCoreConfigCommands.php [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Drupal\az_core\Commands;
+namespace Drupal\az_core\Drush\Commands;
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/AZCoreConfigCommands.php:4]: See https://www.drush.org/12.x/commands/ Tips

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+### Requirements
+
+- Drupal 10
+- Drush 12
+
+### Installation
+
+This module is automatically included in the az_quickstart install profile and enabled by default. These custom Drush commands will work on any installed Quickstart 2 website.
+
+## AZ Core Config Commands
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:13]:
```suggestion
## AZ Core Config Command: Add Permissions
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+AZ Core provides Drush commands to help with managing an AZ Quickstart website.
+
+### Requirements
+
+- Drupal 10
+- Drush 12
+
+### Installation
+
+This module is automatically included in the az_quickstart install profile and enabled by default. These custom Drush commands will work on any installed Quickstart 2 website.
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:11]:
```suggestion
This module is automatically included in the az_quickstart install profile and enabled by default. 
These custom Drush commands will work on any installed Quickstart 2 website.
```

**Comment by bberndt-uaz** [comment]
Tested locally.

#### Standard Quickstart installation
```
 ------------------------ ------------------------ ------- ---------------------- 
  Entity type              Bundle                   Count   Entity Type Provider  
 ------------------------ ------------------------ ------- ---------------------- 
  block                    block                    47      block                 
  block_content_type       block_content_type       3       block_content         
  config_snapshot          config_snapshot          75      config_snapshot       
  crop_type                crop_type                8       crop                  
  editor                   editor                   2       editor                
  embed_button             embed_button             3       embed                 
  field_config             field_config             84      field                 
  field_storage_config     field_storage_config     62      field                 
  filter_format            filter_format            5       filter                
  google_tag_container     google_tag_container     1       google_tag            
  image_style              image_style              27      image                 
  linkit_profile           linkit_profile           2       linkit                
  media_type               media_type               5       media                 
  metatag_defaults         metatag_defaults         7       metatag               
  node_type                node_type                4       node                  
  path_alias               path_alias               1       path_alias            
  redirect                 redirect                 1       redirect              
  responsive_image_style   responsive_image_style   1       responsive_image      
  search_page              search_page              4       search                
  shortcut                 default                  2       shortcut              
  shortcut_set             shortcut_set             1       shortcut              
  smart_date_format        smart_date_format        4       smart_date            
  action                   action                   30      system                
  menu                     menu                     11      system                
  taxonomy_vocabulary      taxonomy_vocabulary      7       taxonomy              
  tour                     tour                     3       tour                  
  user_role                user_role                7       user                  
  user                     user                     2       user                  
  workflow                 workflow                 1       workflows             
  pathauto_pattern         pathauto_pattern         6       pathauto              
  xmlsitemap               xmlsitemap               1       xmlsitemap            
  view                     view                     22      views                 
  paragraphs_type          paragraphs_type          9       paragraphs            
  base_field_override      base_field_override      5       core                  
  entity_view_display      entity_view_display      72      core                  
  entity_form_display      entity_form_display      28      core                  
  entity_form_mode         entity_form_mode         2       core                  
  entity_view_mode         entity_view_mode         47      core                  
  date_format              date_format              13      core                  
 ------------------------ ------------------------ ------- ---------------------- 
```
</details>

#### After installing Quickstart demo content
```
 ------------------------ -------------------------------- ------- ---------------------- 
  Entity type              Bundle                           Count   Entity Type Provider  
 ------------------------ -------------------------------- ------- ---------------------- 
  block                    block                            47      block                 
  block_content            az_custom_menu_block             3       block_content         
  block_content_type       block_content_type               4       block_content         
  config_snapshot          config_snapshot                  76      config_snapshot       
  crop_type                crop_type                        8       crop                  
  editor                   editor                           2       editor                
  embed_button             embed_button                     3       embed                 
  field_config             field_config                     85      field                 
  field_storage_config     field_storage_config             63      field                 
  file                     file                             41      file                  
  filter_format            filter_format                    5       filter                
  google_tag_container     google_tag_container             1       google_tag            
  image_style              image_style                      27      image                 
  linkit_profile           linkit_profile                   2       linkit                
  media_type               media_type                       5       media                 
  media                    az_image                         35      media                 
  media                    az_remote_video                  6       media                 
  menu_link_content        menu_link_content                34      menu_link_content     
  metatag_defaults         metatag_defaults                 7       metatag               
  migration_group          migration_group                  1       migrate_plus          
  node_type                node_type                        4       node                  
  node                     az_event                         23      node                  
  node                     az_flexible_page                 22      node                  
  node                     az_news                          5       node                  
  node                     az_person                        13      node                  
  path_alias               path_alias                       144     path_alias            
  redirect                 redirect                         1       redirect              
  responsive_image_style   responsive_image_style           1       responsive_image      
  search_page              search_page                      4       search                
  shortcut                 default                          2       shortcut              
  shortcut_set             shortcut_set                     1       shortcut              
  smart_date_format        smart_date_format                4       smart_date            
  smart_date_rule          smart_date_rule                  9       smart_date_recur      
  action                   action                           30      system                
  menu                     menu                             11      system                
  taxonomy_term            az_event_categories              23      taxonomy              
  taxonomy_term            az_news_tags                     4       taxonomy              
  taxonomy_term            az_page_categories               12      taxonomy              
  taxonomy_term            az_person_categories             19      taxonomy              
  taxonomy_term            az_person_categories_secondary   11      taxonomy              
  taxonomy_vocabulary      taxonomy_vocabulary              7       taxonomy              
  tour                     tour                             3       tour                  
  user_role                user_role                        7       user                  
  user                     user                             2       user                  
  workflow                 workflow                         1       workflows             
  pathauto_pattern         pathauto_pattern                 6       pathauto              
  xmlsitemap               xmlsitemap                       1       xmlsitemap            
  view                     view                             22      views                 
  paragraphs_type          paragraphs_type                  9       paragraphs            
  paragraph                az_cards                         4       paragraphs            
  paragraph                az_splitscreen                   21      paragraphs            
  paragraph                az_text                          17      paragraphs            
  paragraph                az_text_background               9       paragraphs            
  paragraph                az_text_media                    14      paragraphs            
  paragraph                az_view_reference                4       paragraphs            
  base_field_override      base_field_override              5       core                  
  entity_view_display      entity_view_display              73      core                  
  entity_form_display      entity_form_display              29      core                  
  entity_form_mode         entity_form_mode                 2       core                  
  entity_view_mode         entity_view_mode                 47      core                  
  date_format              date_format                      13      core                  
 ------------------------ -------------------------------- ------- ---------------------- 
```

`lando drush az-core-config-add-permissions` also worked. (It did not find any missing permissions.)

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+- Drush 12
+
+### Installation
+
+This module is automatically included in the az_quickstart install profile and enabled by default. 
+These custom Drush commands will work on any installed Quickstart 2 website.
+
+## AZ Core Config Command: Add Permissions
+
+AZ Core provides Drush commands to help with managing configuration on AZ Quickstart websites.
```
Comment (bberndt-uaz) [inline modules/custom/az_core/src/Drush/Commands/README.md:16]:
```suggestion
## AZ Core Config: Add Permissions

Provides a Drush command to help with managing configuration on AZ Quickstart websites.
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+Topics:
+drush topic docs:output-formats-filters Output formatters and filters: control the command output
+
+Aliases: `ael`
+
+```
+lando drush az-entity-list:list
+```
```
Comment (bberndt-uaz) [inline modules/custom/az_core/src/Drush/Commands/README.md:60]:
```suggestion
```

**Review by bberndt-uaz — APPROVED** [review]
I made some suggestions for the README so the documentation would be similar for the two commands, but otherwise this looks great!

**Review by bberndt-uaz — APPROVED** [review]

File: composer.json [hunk 10]
```diff
@@ -125,7 +125,14 @@
         "seboettg/citeproc-php": "2.6.2"
     },
     "require-dev": {
-        "az-digital/az-quickstart-dev": "~1"
+        "az-digital/az-quickstart-dev": "~1",
+         "drush/drush": ">=12"
```
Comment (joeparsons) [inline composer.json:129]:
```suggestion
        "drush/drush": ">=12"
```
I think I'd prefer to keep the dev dependencies in the dev metapackage.  Maybe we should add the drush requirement there?

Comment (camikazegreen) [reply composer.json:129]: I might be misunderstanding the request here, but I do think that both of the Drush commands affected by this PR will be helpful for live sites and not just during development.

**Comment by trackleft** [comment]
Add Drush as a dependency instead of a dev dependency to az_quickstart composer.json

drush:drush: => 12.4.3

File: composer.json [hunk 11]
```diff
@@ -127,6 +127,12 @@
     "require-dev": {
         "az-digital/az-quickstart-dev": "~1"
     },
+    "suggest": {
+        "drush/drush": ">=12"
+    },
+    "conflict": {
+        "drush/drush": "<12.0"
```
Comment (trackleft) [inline composer.json:134]:
```suggestion
        "drush/drush": "<12.4.3"
```

**Comment by trackleft** [comment]
image

File: composer.json [hunk 12]
```diff
@@ -127,6 +127,12 @@
     "require-dev": {
         "az-digital/az-quickstart-dev": "~1"
```
Comment (trackleft) [inline composer.json:128]: See https://github.com/drupal/core/blob/10.2.x/composer.json#L51C1-L54C1

File: modules/custom/az_core/src/Drush/Commands/AZEntityListCommands.php [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+      foreach ($bundle_types as $bundle) {
+        $count = $this->getEntityCount($entity_type, $bundle);
+        if ($count === 0) {
+          continue;
+        }
+        $all_results[] = [
+          'entity_type' => $entity_type,
+          'bundle' => $bundle,
+          'count' => $count,
+          'module' => $entity_provider,
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/AZEntityListCommands.php:83]: I think this should change to 
```suggestion
          'entity_type_provider' => $entity_type_provider,
```
In order to make way for `$bundle_type_provider`

File: modules/custom/az_core/src/Drush/Commands/AZEntityListCommands.php [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+    'entity_type',
+    'bundle',
+    'count',
+    'module',
+  ])]
+  #[CLI\FieldLabels(labels: [
+    'entity_type' => 'Entity type',
+    'bundle' => 'Bundle',
+    'count' => 'Count',
+    'module' => 'Entity Type Provider',
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/AZEntityListCommands.php:61]:
```suggestion
    'entity_type_provider' => 'Entity Type Provider',
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+### Usage
+
+This module provides the following Drush command:
+`drush az-entity-list:list`
+This command will list entities on an Arizona Quickstart site.
+
+```
+Options:
+ --format[=FORMAT] Format the result data. Available formats: csv,json,list,null,php,print-r,sections,string,table,tsv,var_dump,var_export,xml,yaml [default: table]
+ --fields=FIELDS   Available fields: Entity type (entity_type), Bundle (type), Count (count) [default: entity_type,type,count]
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:38]:
```suggestion
 --fields=FIELDS   Available fields: Entity type (entity_type), Bundle (bundle), Entity Type Provider (entity_type_provider), Count (count), [default: entity_type,bundle,entity_type_provider,count]
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+
+### Output
+
+The default output will count each entity type by bundle, and output in a table by default.
+
+To output in other formats, use `drush az-entity-list:list --help`
+
+```
+ ------------------------ -------------------------------- ------------------- -------
+  Entity type              Bundle                           Provider            Count
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:56]:
```suggestion
  Entity type              Bundle                           Entity Type Provider            Count
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+  paragraph                az_view_reference                paragraphs          4
+  base_field_override      base_field_override              core                5
+  date_format              date_format                      core                13
+  entity_form_display      entity_form_display              core                29
+  entity_form_mode         entity_form_mode                 core                2
+  entity_view_display      entity_view_display              core                73
+  entity_view_mode         entity_view_mode                 core                47
+ ------------------------ -------------------------------- ------------------- -------
+
+```
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:121]:
```suggestion
```
 ------------------------ -------------------------------- ------------------- -------
  Entity type              Bundle                           Entity Type Provider Count
 ------------------------ -------------------------------- ------------------- -------
  block                    block                            block                47
  block_content            az_custom_menu_block             block_content        3
  block_content_type       block_content_type               block_content        4
  config_snapshot          config_snapshot                  config_snapshot      76
  crop_type                crop_type                        crop                 8
  editor                   editor                           editor               2
  embed_button             embed_button                     embed                3
  field_config             field_config                     field                85
  field_storage_config     field_storage_config             field                63
  file                     file                             file                 41
  filter_format            filter_format                    filter               5
  google_tag_container     google_tag_container             google_tag           1
  image_style              image_style                      image                27
  linkit_profile           linkit_profile                   linkit               2
  media_type               media_type                       media                5
  media                    az_image                         media                35
  media                    az_remote_video                  media                6
  menu_link_content        menu_link_content                menu_link_content    34
  metatag_defaults         metatag_defaults                 metatag              7
  migration_group          migration_group                  migrate_plus         1
  node_type                node_type                        node                 4
  node                     az_event                         node                 23
  node                     az_flexible_page                 node                 22
  node                     az_news                          node                 5
  node                     az_person                        node                 13
  path_alias               path_alias                       path_alias           144
  redirect                 redirect                         redirect             1
  responsive_image_style   responsive_image_style           responsive_image     1
  search_page              search_page                      search               4
  shortcut                 default                          shortcut             2
  shortcut_set             shortcut_set                     shortcut             1
  smart_date_format        smart_date_format                smart_date           4
  smart_date_rule          smart_date_rule                  smart_date_recur     9
  action                   action                           system               30
  menu                     menu                             system               11
  taxonomy_vocabulary      taxonomy_vocabulary              taxonomy             7
  taxonomy_term            az_event_categories              taxonomy             23
  taxonomy_term            az_news_tags                     taxonomy             4
  taxonomy_term            az_page_categories               taxonomy             12
  taxonomy_term            az_person_categories             taxonomy             19
  taxonomy_term            az_person_categories_secondary   taxonomy             11
  tour                     tour                             tour                 3
  user_role                user_role                        user                 7
  user                     user                             user                 2
  workflow                 workflow                         workflows            1
  pathauto_pattern         pathauto_pattern                 pathauto             6
  xmlsitemap               xmlsitemap                       xmlsitemap           1
  view                     view                             views                22
  paragraphs_type          paragraphs_type                  paragraphs           9
  paragraph                az_cards                         paragraphs           4
  paragraph                az_splitscreen                   paragraphs           21
  paragraph                az_text                          paragraphs           17
  paragraph                az_text_background               paragraphs           9
  paragraph                az_text_media                    paragraphs           14
  paragraph                az_view_reference                paragraphs           4
  base_field_override      base_field_override              core                 5
  date_format              date_format                      core                 13
  entity_form_display      entity_form_display              core                 29
  entity_form_mode         entity_form_mode                 core                 2
  entity_view_display      entity_view_display              core                 73
  entity_view_mode         entity_view_mode                 core                 47
 ------------------------ -------------------------------- ------------------- -------

```
```

File: composer.json [hunk 18]
```diff
@@ -127,6 +127,12 @@
...
     },
+    "suggest": {
+        "drush/drush": ">=12.4.3"
```
Comment (joeparsons) [inline composer.json:131]: `suggest` doesn't support version constraints.

https://getcomposer.org/doc/04-schema.md#suggest

>The format is like package links above, except that the values are free text and not version constraints.

Comment (trackleft) [inline composer.json:131]:
```suggestion
```

File: composer.json [hunk 19]
```diff
@@ -127,6 +127,11 @@
     "require-dev": {
         "az-digital/az-quickstart-dev": "~1"
     },
+    "suggest": {
+    },
+    "conflict": {
+        "drush/drush": "<12.4.3"
+    },
```
Comment (trackleft) [inline composer.json:134]:
```suggestion
    "conflict": {
        "drush/drush": "<12.4.3"
    },
```

File: modules/custom/az_core/src/Drush/Commands/AZEntityListCommands.php [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+
+    public function list(array $options = ['format' => 'table']): RowsOfFields {
+    $all_results = [];
+    $entity_types = array_keys($this->entityTypeManager->getDefinitions());
+
+    foreach ($entity_types as $entity_type) {
+      // If the entity has bundle types add them to the list. Otherwise, just
+      // use the entity type as the bundle.
+      $bundle_types = $this->getBundleTypes($entity_type);
+      $entity_provider = $this->entityTypeManager->getDefinition($entity_type)->getProvider();
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/AZEntityListCommands.php:73]:
```suggestion
      $entity_type_provider = $this->entityTypeManager->getDefinition($entity_type)->getProvider();
```

**Review by joeparsons — APPROVED** [review]

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+This module provides the following Drush command:
+`drush az-entity-list:list`
+This command will list entities on an Arizona Quickstart site.
+
+```
+Options:
+ --format[=FORMAT] Format the result data. Available formats: csv,json,list,null,php,print-r,sections,string,table,tsv,var_dump,var_export,xml,yaml [default: table]
+ --fields=FIELDS   Available fields: Entity type (entity_type), Bundle (bundle), Entity Type Provider (entity_type_provider), Count (count), [default: entity_type,bundle,entity_type_provider,count]
+ --field=FIELD     Select just one field, and force format to *string*.
+ --filter[=FILTER] Filter output based on provided expression
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:40]:
```suggestion
```

File: modules/custom/az_core/src/Drush/Commands/README.md [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+Topics:
+drush topic docs:output-formats-filters Output formatters and filters: control the command output
```
Comment (trackleft) [inline modules/custom/az_core/src/Drush/Commands/README.md:44]:
```suggestion
```

File: modules/custom/az_core/src/Drush/Commands/README.md (hunk 21 above)
Comment (trackleft) [reply modules/custom/az_core/src/Drush/Commands/README.md:40]: Seems filter is not yet supported. Sorry.

**Review by joeparsons — APPROVED** [review]

**Comment by camikazegreen** [comment]
Are you planning on merging any of your commit suggestions before I put my review on this?

**Review by camikazegreen — APPROVED** [review]
Tested in the Probo ssh shell by following the Readme. 
Everything looks great for both the default option and the csv format.

**Review by tadean — APPROVED** [review]

**Pull request merged**